main/src/hadd.cxx

```diff
@@ -376,33 +376,51 @@ int main( int argc, char **argv )
    if (maxopenedfiles > 0) {
       fileMerger.SetMaxOpenedFiles(maxopenedfiles);
    }
-   if (newcomp == -1) {
-      if (useFirstInputCompression || keepCompressionAsIs) {
-         // grab from the first file.
-         TFile *firstInput = nullptr;
-         if (argv[ffirst] && argv[ffirst][0]=='@') {
-            std::ifstream indirect_file(argv[ffirst]+1);
-            if( ! indirect_file.is_open() ) {
-               std::cerr<< "hadd could not open indirect file " << (argv[ffirst]+1) << std::endl;
-               return 1;
-            }
-            std::string line;
-            while( indirect_file ){
-               if( std::getline(indirect_file, line) && line.length() ) {
-                  firstInput = TFile::Open(line.c_str());
-                  break;
+   std::vector<std::string> allSubfiles;
```

Comment:
I believe it would help to have just a small comment at the beginning and end of this section to signify that here we are only accumulating the list of input files from the indirect file

Reply:
Done